### PR TITLE
Adding ST7735 TFT Controller driver.

### PR DESCRIPTION
**Pull request by Jacob-Scheiffler**

### CODE REVIEW

File: Display/tft_st7735.c [hunk 1]
```diff
@@ -0,0 +1,698 @@
...
+
+static tft_rotation_t tft_rotation = ROTATE_0;
+static unsigned int g_foreground_color;
+static unsigned int g_background_color;
+static unsigned char *g_font;
+
+/********************************* Static Functions **************************/
+static void write_command(unsigned char command)
+{
+    TFT_SPI_Write(&command, 1, false);
```
Comment (Jake-Carter) [inline Display/tft_st7735.c:84]: Looks good.  I just did a similar thing yesterday for the NewHaven drivers.  I would suggest adding `extern` definitions for these into the header file.

Ex:  https://github.com/Analog-Devices-MSDK/Libraries-miscDrivers/blob/fe4c6c4c61fab1d518c8e2f3ceb3033390c29996/Display/tft_st7789v.h#L152

I will change my function names to match yours.  We should also discuss standardizing the SPI write headers.  Some of the older drivers had split functions for writing a single byte vs transmitting from a buffer.

Let's merge these in for now (with the added `extern` definitions) and standardize the SPI transport layer in a later PR.

**Review by Jake-Carter — CHANGES_REQUESTED** [review]
Looks good!  Approved pending the one comment below.

**Review by lorne-maxim — APPROVED** [review]

**Comment by lorne-maxim** [comment]
Ignoring the CPP lint errors for now.  They are all header-guard errors that will pass once this submodule is absorbed into the msdk repo.

**Pull request merged**